packages/components/src/number-control/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import BaseControl from '../base-control';
+
+function NumberControl( {
+	className,
+	label,
+	value,
+	instanceId,
+	id = `inspector-number-control-${ instanceId }`,
+	onChange,
+	help,
+	numberUnit,
+	...props
+} ) {
+	const onChangeValue = ( event ) => onChange( Number( event.target.value ) );
```

Comment:
Maybe we can change this line to:
```
	const onChangeValue = ( event ) => onChange(
		event.target.value === '' ? undefined : Number( event.target.value )
	);
```

The current version causes a bug -- it becomes impossible to have the field empty. It is possible to reproduce the bug testing the spacer.
The bug happens because Number('') === 0;